From ff248f6a2c7db1d5bf65e1b2de77122525d609ea Mon Sep 17 00:00:00 2001
From: Yan Markman <ymarkman@marvell.com>
Date: Wed, 3 Jan 2018 21:33:04 +0200
Subject: [PATCH 2203/2241] dts: a8040: update phy settings of Armada-8040 DB
 for setups B, D, E

Update PHY configuration for Armada-8040 DB board, for setups B, D, E
  on the cp1_emac0:
a8040-B: SGMII0 to SFI
a8040-D: SFI to SGMII0
a8040-E: SFI to 1000base-x (HS_SGMII0 speed 2500)

Resulting NET-domain configuration is:
=======:--- A ----- | --- B ---- | --- D ---- | --- E ----
  cp0:0  eth0 sfi      -x-         eth0 sfi     eth0 sfi
  cp0:3  eth1 rgmii   eth0 rgmii   eth1 rgmii    -x-
>>cp1:0  eth2 sfi     eth1 sfi     eth2 sgmii   eth1 1000base-x HS=2500
  cp1:2  eth3 rgmii   eth2 rgmii    -x-         eth2 rgmii

Change-Id: Ib066c81736af308cab0d34a267d647d7e1111d6f
Signed-off-by: Yan Markman <ymarkman@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/48527
Reviewed-by: Omri Itach <omrii@marvell.com>
Tested-by: Omri Itach <omrii@marvell.com>
---
 Documentation/mvebu/a7k-a8k/armada8040-db-setup.txt | 6 +++---
 arch/arm64/boot/dts/marvell/armada-8040-db-B.dts    | 4 ++--
 arch/arm64/boot/dts/marvell/armada-8040-db-D.dts    | 4 ++--
 arch/arm64/boot/dts/marvell/armada-8040-db-E.dts    | 5 +++--
 4 files changed, 10 insertions(+), 9 deletions(-)

diff --git a/Documentation/mvebu/a7k-a8k/armada8040-db-setup.txt b/Documentation/mvebu/a7k-a8k/armada8040-db-setup.txt
index c111f4d..fd58e20 100644
--- a/Documentation/mvebu/a7k-a8k/armada8040-db-setup.txt
+++ b/Documentation/mvebu/a7k-a8k/armada8040-db-setup.txt
@@ -147,7 +147,7 @@ Network configuration
  Interface	| CPN#	| PPv2 Port	| GMAC Port	| Board Interface		|
  ---------------|-------|---------------|---------------|-------------------------------|
  eth0		| 0	| 2		| 3		| RGMII1			|
- eth1		| 1	| 0		| 0		| SGMII				|
+ eth1		| 1	| 0		| 0		| SFI (10G Lane 2 connected to SFP)
  eth2		| 1	| 1		| 2		| RGMII0			|
 
 3: C:
@@ -162,13 +162,13 @@ Network configuration
  ---------------|-------|---------------|---------------|-------------------------------|
  eth0		| 0	| 0		| 0		| 10G (Lane 2 connected to SFP)	|
  eth1		| 0	| 2		| 3		| RGMII1			|
- eth2		| 1	| 0		| 0		| 10G (Lane 2 connected to SFP)	|
+ eth2		| 1	| 0		| 0		| SGMII0			|
 
 5: E:
  Interface	| CPN#	| PPv2 Port	| GMAC Port	| Board Interface		|
  ---------------|-------|---------------|---------------|-------------------------------|
  eth0		| 0	| 0		| 0		| 10G (Lane 2 connected to SFP)	|
- eth1		| 1	| 0		| 0		| 10G (Lane 2 connected to SFP)	|
+ eth1		| 1	| 0		| 0		| 1000base-x HS_SGMII0, speed=2500
  eth2		| 1	| 1		| 2		| RGMII0			|
 
 Notes:
diff --git a/arch/arm64/boot/dts/marvell/armada-8040-db-B.dts b/arch/arm64/boot/dts/marvell/armada-8040-db-B.dts
index 193c2d0..d7000de 100644
--- a/arch/arm64/boot/dts/marvell/armada-8040-db-B.dts
+++ b/arch/arm64/boot/dts/marvell/armada-8040-db-B.dts
@@ -184,8 +184,8 @@
 			};
 			gop {
 				cp1_emac0: mac0 {
-					phy-mode = "sgmii"; /* lane-2 */
-					phys = <&cp1_comphy 2 COMPHY_SGMII0>;
+					phy-mode = "sfi"; /* lane-2 */
+					phys = <&cp1_comphy 2 COMPHY_SFI>;
 					phy-names = "comphy";
 				};
 				cp1_emac2: mac2 {
diff --git a/arch/arm64/boot/dts/marvell/armada-8040-db-D.dts b/arch/arm64/boot/dts/marvell/armada-8040-db-D.dts
index fbd9ec1..372037b 100644
--- a/arch/arm64/boot/dts/marvell/armada-8040-db-D.dts
+++ b/arch/arm64/boot/dts/marvell/armada-8040-db-D.dts
@@ -198,8 +198,8 @@
 			};
 			gop {
 				cp1_emac0: mac0 {
-					phy-mode = "sfi"; /* lane-2 */
-					phys = <&cp1_comphy 2 COMPHY_SFI>;
+					phy-mode = "sgmii"; /* lane-2 */
+					phys = <&cp1_comphy 2 COMPHY_SGMII0>;
 					phy-names = "comphy";
 				};
 				cp1_emac2: mac2 {
diff --git a/arch/arm64/boot/dts/marvell/armada-8040-db-E.dts b/arch/arm64/boot/dts/marvell/armada-8040-db-E.dts
index 63ce07d..23cadad 100644
--- a/arch/arm64/boot/dts/marvell/armada-8040-db-E.dts
+++ b/arch/arm64/boot/dts/marvell/armada-8040-db-E.dts
@@ -204,8 +204,9 @@
 			};
 			gop {
 				cp1_emac0: mac0 {
-					phy-mode = "sfi"; /* lane-2 */
-					phys = <&cp1_comphy 2 COMPHY_SFI>;
+					phy-mode = "1000base-x"; /* lane-2 */
+					phy-speed = <2500>;
+					phys = <&cp1_comphy 2 COMPHY_HS_SGMII0>;
 					phy-names = "comphy";
 				};
 				cp1_emac2: mac2 {
-- 
2.7.4

